### PR TITLE
Don't use regular expressions for parsing delimited strings

### LINKED ISSUE #10391: Test legacy bundles before release
We will be able to test only in chromium headless, but that will cover quite a few bases. We will have to test manually in IE.

### PR DESCRIPTION
**Pull request by mattpap**
This also addresses some of my concerns from recent "math text" PRs. I also found a glaring bug in `HasProps` equality, which I fixed in this PR, but I still need to investigate a document unit test failure.

fixes #10391

### CODE REVIEW

File: bokehjs/src/lib/core/has_props.ts [hunk 1]
```diff
@@ -230,7 +230,7 @@ export abstract class HasProps extends Signalable() implements Equatable, Printa
   [equals](that: this, cmp: Comparator): boolean {
     for (const p0 of this) {
       const p1 = that.property(p0.attr)
-      if (cmp.eq(p0.get_value(), p1.get_value()))
+      if (!cmp.eq(p0.get_value(), p1.get_value()))
```
Comment (bryevdv) [inline bokehjs/src/lib/core/has_props.ts:233]: ❗

File: bokehjs/src/lib/models/axes/axis.ts [hunk 2]
```diff
@@ -46,46 +47,30 @@ export class AxisView extends GuideRendererView {
...
 
+  protected async _init_axis_label(): Promise<void> {
+    const {axis_label} = this.model
     if (axis_label != null) {
-      if (axis_label instanceof MathText) {
-        this.axis_label_math_text_view = await build_view(axis_label, {parent: this})
-      } else if (is_tex_string(axis_label)) {
-        const math_text = tex_from_text_like(axis_label)
-        if (math_text instanceof TeX) {
-          this.model.axis_label = math_text
```
Comment (mattpap) [inline bokehjs/src/lib/models/axes/axis.ts:67]: Updating model's properties was one of my concerns. The original would require multiple server updates per plot and would interfere with user's set values.

File: bokehjs/src/lib/core/has_props.ts (hunk 1 above)
Comment (mattpap) [reply bokehjs/src/lib/core/has_props.ts:233]: A side effect of not having unit tests for this case.

File: bokehjs/src/lib/models/axes/axis.ts (hunk 2 above)
Comment (bryevdv) [reply bokehjs/src/lib/models/axes/axis.ts:67]: I wouldn't mind splitting `parse_delimited_string` delimited string up too, into a "these are the math parts of the string" part and "turn these math parts into Bokeh models" part, but, later.

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
tested locally, examples and docs

File: bokehjs/src/lib/models/axes/axis.ts (hunk 2 above)
Comment (mattpap) [reply bokehjs/src/lib/models/axes/axis.ts:67]: This will happen next, i.e. composite typeset text and other (math, maybe graphics) content.

**Pull request merged**